### PR TITLE
Get go stdlib version from go.mod

### PR DESCRIPTION
**Pull request by hogo6002**
Fix https://github.com/google/osv-scanner/issues/679

### CODE REVIEW

File: pkg/lockfile/parse-go-lock.go [hunk 1]
```diff
@@ -82,6 +82,20 @@ func (e GoLockExtractor) Extract(f DepFile) ([]PackageDetails, error) {
 		}
 	}
 
+	if parsedLockfile.Go != nil && parsedLockfile.Go.Version != "" {
+		goVersion := parsedLockfile.Go.Version
+		versionParts := strings.Split(goVersion, ".")
```
Comment (cuixq) [inline pkg/lockfile/parse-go-lock.go:87]: maybe we should use some Go package that deals with the versioning here?
I found this package: https://pkg.go.dev/golang.org/x/mod/semver, but this requires a "v" prefix in the version string.

Comment (G-Rath) [reply pkg/lockfile/parse-go-lock.go:87]: If only we had a robust semantic based version parsing and comparing library already... 😁

(`internal/semantic` should have everything you need for this)

Comment (hogo6002) [reply pkg/lockfile/parse-go-lock.go]: Updated, thanks!

**Review by another-rex — APPROVED** [review]
LGTM

**Pull request merged**